src/utils/circuit-to-svg.tsx

```diff
@@ -0,0 +1,317 @@
+import jscad from "@jscad/modeling"
+import type { AnySoupElement } from "@tscircuit/soup"
+import { su } from "@tscircuit/soup-util"
+import { Footprinter3d } from "jscad-electronics"
+import { convertCSGToThreeGeom, createJSCADRenderer } from "jscad-fiber"
+import { executeJscadOperations, jscadPlanner } from "jscad-planner"
+import { JSDOM } from "jsdom"
+import * as THREE from "three"
+import { BufferGeometry, Float32BufferAttribute } from "three"
+import { OBJLoader } from "three/examples/jsm/loaders/OBJLoader.js"
+import { STLLoader } from "three/examples/jsm/loaders/STLLoader.js"
+import { SVGRenderer } from "three/examples/jsm/renderers/SVGRenderer.js"
+import { createBoardGeomFromSoup } from "../soup-to-3d"
+
+interface CircuitToSvgOptions {
+  width?: number
+  height?: number
+  viewAngle?: "top" | "isometric" | "front" | "side"
+  backgroundColor?: string
+  padding?: number
+  zoom?: number
+}
+
+// Setup JSDOM and globals needed for THREE.js
+const dom = new JSDOM()
+global.document = dom.window.document
+global.window = dom.window as any
+global.navigator = dom.window.navigator
+global.Element = dom.window.Element
+global.HTMLElement = dom.window.HTMLElement
```

Comment:
make sure to remove jsdom in the final implementation, it's not supposed to be added by libraries